plugins/modules/list_clusters.py

```diff
@@ -47,15 +51,37 @@
 from ansible.module_utils.basic import AnsibleModule
 
 BASE_URL = "https://api.openshift.com/api/assisted-install/v2/clusters"
+# add additional query parameters to the query_params_list
+QUERY_PARAMS_LIST = ["with_hosts"]
+
+def get_query_param_str(module):
+    query_params_dict = {}
+    for key in QUERY_PARAMS_LIST:
+        val = module.params.get(key)
+        if val:
+            query_params_dict[key] = val
+
+    query_param_str = ""
+    dict_len = len(query_params_dict)
+    if dict_len != 0:
+        query_param_str += "?"
+        i = 0
+        for k in query_params_dict:
+            if i != 0:
+                query_param_str += "&"
+            query_param_str += str(k) + "=" + str(query_params_dict[k])
+            i = i + 1
+    return query_param_str
 
 def run_module():
     module_args = dict(
         fail=dict(type="bool", required=False, default=False),
+        # any API query parameters may have to be added here
+        with_hosts=dict(type="bool", required=False, default=False),
     )
     token = os.environ.get('AI_API_TOKEN')
     module = AnsibleModule(argument_spec=module_args, supports_check_mode=False)
-
-    response = requests.get( BASE_URL, headers={'Content-Type': 'application/json', 'Authorization': f'Bearer {token}'} )
+    response = requests.get( BASE_URL + get_query_param_str(module), headers={'Content-Type': 'application/json', 'Authorization': f'Bearer {token}'} )
```

Comment:
You can avoid the reimplementation of the parameters by using `params`[1] from requests. The value passed is a dict with `key: value`

It would look like this:
```suggestion
    params = {}
    if module.params['with_hosts']:
        params = params | {"with_hosts": "true"}

    response = requests.get( BASE_URL, params=params, headers={'Content-Type': 'application/json', 'Authorization': f'Bearer {token}'} )
```

[1] https://docs.python-requests.org/en/latest/user/quickstart/#passing-parameters-in-urls

Reply:
@tonyskapunk Great suggestion, was not aware that I could pass params as separate parameter to the requests.get() function.